lib/appium.js

```diff
@@ -178,24 +183,30 @@ class AppiumDriver extends BaseDriver {
       }
     }
 
-    let runningDriversData;
-    try {
-      runningDriversData = this.curSessionDataForDriver(InnerDriver);
-    } catch (e) {
-      throw new errors.SessionNotCreatedError(e.message);
-    }
-
-    let innerSessionId, dCaps;
+    let runningDriversData, otherPendingDriversData;
     let d = new InnerDriver(this.args);
-    try {
+    await sessionCreationLock.acquire(AppiumDriver.name, async () => {
```

Comment:
i'm a little confused why we need to lock this block; we're not doing anything asynchronous in here right?